contracts/rewards/boosted-staking/BoostedDualVault.sol

```diff
@@ -0,0 +1,705 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.8.2;
+
+// Internal
+import { IRewardsRecipientWithPlatformToken } from "../../interfaces/IRewardsDistributionRecipient.sol";
+import { IBoostedDualVaultWithLockup } from "../../interfaces/IBoostedDualVaultWithLockup.sol";
+import { IRewardsDistributionRecipient, InitializableRewardsDistributionRecipient } from "../InitializableRewardsDistributionRecipient.sol";
+import { BoostedTokenWrapper } from "./BoostedTokenWrapper.sol";
+import { PlatformTokenVendor } from "../staking/PlatformTokenVendor.sol";
+import { Initializable } from "../../shared/@openzeppelin-2.5/Initializable.sol";
+
+// Libs
+import { IERC20, SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { StableMath } from "../../shared/StableMath.sol";
+
+/**
+ * @title  BoostedVault
+ * @author mStable
+ * @notice Accrues rewards second by second, based on a users boosted balance
+ * @dev    Forked from rewards/staking/StakingRewards.sol
+ *         Changes:
+ *          - Lockup implemented in `updateReward` hook (20% unlock immediately, 80% locked for 6 months)
+ *          - `updateBoost` hook called after every external action to reset a users boost
+ *          - Struct packing of common data
+ *          - Searching for and claiming of unlocked rewards
+ */
+contract BoostedDualVault is
+    IBoostedDualVaultWithLockup,
+    IRewardsRecipientWithPlatformToken,
+    Initializable,
+    InitializableRewardsDistributionRecipient,
+    BoostedTokenWrapper
+{
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+    using SafeCast for uint256;
+
+    event RewardAdded(uint256 reward, uint256 platformReward);
+    event Staked(address indexed user, uint256 amount, address payer);
+    event Withdrawn(address indexed user, uint256 amount);
+    event Poked(address indexed user);
+    event RewardPaid(address indexed user, uint256 reward, uint256 platformReward);
+
+    /// @notice token the rewards are distributed in. eg MTA
+    IERC20 public immutable rewardsToken;
+    /// @notice token the platform rewards are distributed in. eg WMATIC
+    IERC20 public immutable platformToken;
+    /// @notice contract that holds the platform tokens
+    PlatformTokenVendor public platformTokenVendor;
+    /// @notice total raw balance
+    uint256 public totalRaw;
+
+    uint64 public constant DURATION = 7 days;
```

Comment:
Let's make `DURATION` immutable and set in the constructor so we have some flexibility on deployment on the period lengths

Reply:
Personally I would rather keep all reward contracts with the same distribution, this should make automating them easier in the future, as everything will happen on a weekly cadence. I think it will be easier to manage. Can certainly make it an immutable and use it if there is a strong reason to in future

Reply:
ok, let's leave it as a constant. No need to change to an immutable.